tests/test_read.py

```diff
@@ -16,6 +18,30 @@
     has_hdf = False
 
 
+# Fixture to create test datasets within temporary directory
+@pytest.fixture(scope='function')
+def test_data_dir(tmpdir):
```

Comment:
Thoughts about changing it from a fixture to a regular function that you pass a `pathlib.Path` object to the output raster path instead? Then, you could use the `tmp_path` fixture in the test function that controls where the raster is saved. It could also open up opportunities to further generalize it in the future.

Reply:
I'm afraid I don't quite understand your suggestion here and why it would be preferred over a fixture.

Reply:
Sometimes it is easier to communicate with code :smile:. Here is an example of what I mean: https://github.com/corteva/rioxarray/blob/ce1194acb4ababbb4c6570d72cf64aa2760c0fa2/test/integration/test_integration__io.py#L376-L420

Reply:
How about something like this? https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#factories-as-fixtures

Reply:
Looks good :+1: